File: Adm/Models/Clientes.cs
using Newtonsoft.Json;
using System;
using System.Collections.Generic;
using System.IO;

namespace Adm.Models
{
    // Classe Clientes herda de CRUD<Cliente>
    public class Clientes : CRUD<Cliente>
    {

        // protected List<Cliente> objetos = new List<Cliente>();
        // Método para salvar os clientes no arquivo JSON
        public override void Salvar()
        {
            if (!Directory.Exists("../../Data"))
            {
                Directory.CreateDirectory("../../Data");
            }

            try
            {
                using (StreamWriter arquivo = new StreamWriter("../../Data/cliente.json"))
                {
                    var dados = new List<Dictionary<string, object>>();
                    foreach (var cliente in objetos)
                    {
                        dados.Add(cliente.ToDict());  // Convertendo o cliente para dicionário
                    }

                    // Serializar a lista de dicionários e gravar no arquivo JSON
                    string json = JsonConvert.SerializeObject(dados);
                    arquivo.Write(json);
                }
            }
            catch (IOException e)
            {
                Console.WriteLine(e.Message);
            }
        }

        // Sobrescrevendo o método abrir
        public override void Abrir()
        {
            try
            {
                using (StreamReader reader = new StreamReader("../../Data/cliente.json"))
                {
                    string json = reader.ReadToEnd();  // Lê o conteúdo do arquivo JSON
                    var listaClientes = JsonConvert.DeserializeObject<List<Dictionary<string, object>>>(json);

                    if (listaClientes != null)
                    {
                        objetos.Clear();
                        foreach (var obj in listaClientes)
                        {
                            // Cria o objeto Cliente a partir do dicionário
                            var cliente = new Cliente(
                                Convert.ToInt32(obj["id"]),
                                obj["nome"].ToString(),
                                obj["email"].ToString(),
                                obj["fone"].ToString(),
                                obj["senha"].ToString(),
                                Convert.ToBoolean(obj["adm"])
                            );

                            objetos.Add(cliente);  // Adiciona o cliente à lista
                        }
                    }
                }
            }
            catch (FileNotFoundException)
            {
                // Não faz nada se o arquivo não for encontrado
            }
            catch (IOException e)
            {
        
            }
        }
    }
}

File: Adm/Models/src/Clientes.java
package Adm.Models.src;

import com.google.gson.Gson;
import com.google.gson.reflect.TypeToken;

import java.io.FileReader;
import java.io.FileWriter;
import java.io.FileNotFoundException;
import java.io.IOException;
import java.lang.reflect.Type;
import java.util.ArrayList;
import java.util.List;

public class Clientes extends CRUD<Cliente> {


    @Override
    public void salvar() {
        try {
            FileWriter writer = new FileWriter("Data/cliente.json");
            Gson gson = new Gson();
            gson.toJson(objetos, writer); // Serializa os objetos e escreve no arquivo
            writer.close();
        } catch (IOException e) {
            e.printStackTrace();
        }
    }

    @Override
    public void abrir() {
        // Inicializa a lista objetos para evitar NullPointerException
        if (objetos == null) {
            objetos = new ArrayList<>();  // Se objetos for null, inicializa como lista vazia
        }

        try {
            FileReader reader = new FileReader("Data/cliente.json");
            Type listType = new TypeToken<List<Cliente>>(){}.getType();
            // Lê o arquivo JSON e desserializa para a lista de objetos
            List<Cliente> listaClientes = new Gson().fromJson(reader, listType);
            if (listaClientes != null) {
                objetos = listaClientes; // Se o arquivo contiver dados, atualiza a lista objetos
            }
            reader.close();
        } catch (FileNotFoundException e) {
            // Arquivo não encontrado, não há dados para carregar
            objetos = new ArrayList<>(); // Inicializa a lista vazia
        } catch (IOException e) {
            e.printStackTrace();
        }
    }
}